Design/DesignDoc.md

```diff
@@ -0,0 +1,220 @@
+
+String expenseId;
+String expenseName;
+String expenseAmount; 
+String tag;
+ZonedDateTime expenseDate;
+
+// BudgetModel
+
+String budgetId;
+String targetAmount; 
+Boolean status; 
+ZonedDateTime date;
+
+```
+
+## 6.2. _GetExpense Endpoint_
+
+* Accepts `GET` requests to `/expense/:date`
+* Accepts an expenseID and returns the corresponding ExpenseModel.
+    * If the given expenseID is not found, will throw a
+      `ExpenditureNotFoundException`
+
+## 6.3 _GetAllExpenses Endpoint_
+
+* Accepts `GET` requests to `/expense`
+* Returns all the expenses with any tag in the ExpenseModel format.
+    * If there is no data found, will throw a
+      `NoDataFoundException`
+
+## 6.4. GetTaggedExpense Endpoint
+
+* Accepts `GET` requests to `/expense/:tag`
+* Returns all the expenses for the requested tag in the ExpenseModel format.
+    * If there is no department found, will throw a
+      `InvalidDepartmentException`
+
+## 6.5. _UpdateExpense Endpoint_
+
+* Accepts `PUT` requests to `/expense/:date`
+* Accepts data to update an expense including an updated
+  expenseName, expenseAmount, tag, and date. Returns the updated expense.
+* If the expenseID or name is not found, will throw a   
+  `ExpenseNotFoundException`
+* For security concerns, we will validate the provided expense name does not
+  contain invalid characters: `" ' \`
+* If the expense name contains invalid characters, will throw an
+  `InvalidAttributeValueException`
+
+## 6.6. _AddExpense Endpoint_
+
+* Accepts `POST` requests to `/expense/:date`
+* Accepts data to create a new expenditure which includes the
+  expenseId, expenseName, expenseAmount, tag, and date. Returns the new   
+  expense.
+* For security concerns, we will validate the provided expense name does not
+  contain invalid characters: `" ' \`
+* If the expense name contains invalid characters, will throw an
+  `InvalidAttributeValueException`
+
+## 6.7. _GetBudget Endpoint_
+
+* Accepts `GET` requests to `/budget/:date`
+* Accepts a date and returns the corresponding BudgetModel.
+    * If the given date is not found, will throw a
+      `BudgetNotFoundException`
+
+## 6.6. _AddBudget Endpoint_
+
+* Accepts `POST` requests to `/budget/:date`
+* Accepts data to create a new budget which includes the
+  budgetId, targetAmount,and date. Returns the new   
+  budget.
+* For security concerns, we will validate the provided budget name does not
+  contain invalid characters: `" ' \`
+* If the budget name contains invalid characters, will throw an
+  `InvalidAttributeValueException`
+
+## 6.5. _UpdateBudget Endpoint_
+
+* Accepts `PUT` requests to `/budget/:date`
+* Accepts data to update a budget including an updated
+  targetAmount,and date. Returns the updated budget.
+* If the budgetID is not found, will throw a   
+  `BudgetNotFoundException`
+
+## 6.6. _BudgetStatus Endpoint_
+
+* Accepts `GET` requests to `/budget/:date`
+* Accepts a date and returns the status (achieved/ not achieved) of specified budget.
```

Comment:
Re: my comment above on the `status` field - I would have this endpoint query the budget and corresponding expenses to see if it's achieved/not achieved vs having a set field.